lib/bap_disasm/bap_disasm_symtab.ml

```diff
@@ -52,19 +52,27 @@ let merge m1 m2 =
   Memmap.to_sequence m2 |> Seq.fold ~init:m1 ~f:(fun m1 (mem,x) ->
       Memmap.add m1 mem x)
 
+let filter_mem mem name entry =
+  Memmap.filter mem ~f:(fun (n,e,_) ->
+      not(String.(name = n) || Block.(entry = e)))
+
 let remove t (name,entry,_) : t = {
   names = Map.remove t.names name;
   addrs = Map.remove t.addrs (Block.addr entry);
-  memory = Memmap.filter t.memory ~f:(fun (n,e,_) ->
-      not(String.(name = n) || Block.(entry = e)))
+  memory = filter_mem t.memory name entry;
 }
 
+let filter t ((name,entry,_ ) as fn) =
```

Comment:
OK, so from the IRL discussion, the remove function is inefficient since it filters the memory mapping even if the symbol is not in the symtab, that fact that could be checked very fast with Set membership operation.

However, this is the problem of the `remove` function, that should be fixed on its side. So, please remove the filter function and add the corresponding check to the `remove` function (in fact a single test for inclusion in `addrs` would be sufficient.